File: i.cs
using System;
class Program{
  static string[] input = Console.ReadLine().Split(' ');
  static double N = double.Parse(input[0]);
  static double M = double.Parse(input[1]);

  static bool isLarger(double x){
    double result = (N + 1) * Math.Pow(x, 5) + Math.Pow(x, 4) + Math.Pow(x, 3) + Math.Pow(x, 2) + Math.Pow(x, 1) + 1;
    return result > M;
  }

  static void Main(string[] args){
    double left = 0;
    double right = 100000;
    double mid = (left + right) / 2;
    while (right - left > 0.01){
      if(isLarger(mid)){
        right = mid;
      }
      else{
        left = mid;
      }
      mid = (left + right) / 2;   
    }

    Console.WriteLine(mid);
  }
}

File: i.kt
import kotlin.math.*
fun main() {
  val (N, M) = readLine()!!.split(" ").map{it.toInt()}

  fun isLarger(x: Double): Boolean {
    var result = (N + 1) * x.pow(5) + x.pow(4) + x.pow(3) + x.pow(2) + x.pow(1) + 1
    return result > M
  }

  var left = 1.0
  var right = 100000.0
  var mid = (left + right) / 2  

  while(right - left > 0.01){        
    if(isLarger(mid)) right = mid
    else left = mid
    mid = (left + right) / 2    
  }

    println(mid)
}